lib/reversetunnel/agent.go

```diff
@@ -260,22 +259,43 @@ func (a *Agent) getHostCheckers() ([]ssh.PublicKey, error) {
 
 // getReverseTunnelDetails pings the remote Teleport Proxy address in order to check if this is Web Service or ReverseTunnel Service address.
 // If this is Web Service port check if proxy support ALPN SNI Listener.
-func (a *Agent) getReverseTunnelDetails() *reverseTunnelDetails {
+func (a *Agent) getReverseTunnelDetails(addr string) *reverseTunnelDetails {
 	pd := reverseTunnelDetails{TLSRoutingEnabled: false}
-	resp, err := webclient.Find(a.ctx, a.Addr.Addr, lib.IsInsecureDevMode(), nil)
+	resp, err := webclient.Find(a.ctx, addr, lib.IsInsecureDevMode(), nil)
 	if err != nil {
 		// If TLS Routing is disabled the address is the proxy reverse tunnel
 		// address the ping call will always fail.
-		a.log.Infof("Failed to ping web proxy %q addr: %v", a.Addr.Addr, err)
+		a.log.Infof("Failed to ping web proxy %q addr: %v", addr, err)
 	} else {
 		pd.TLSRoutingEnabled = resp.Proxy.TLSRoutingEnabled
 	}
 	return &pd
 }
 
+func (a *Agent) addr() (*utils.NetAddr, error) {
```

Comment:
Each agent only ever connects once, and is removed if it fails.  We should resolve the address once per agent, and then return the same address on each subsequent call, otherwise the agent may report a different address than the one it is actually connected to.